File: src/main/java/kitchenpos/eatinorders/tobe/OrderTable.java
package kitchenpos.eatinorders.tobe;

import java.util.UUID;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class OrderTable {

    @Id
    private UUID id;

    @Embedded
    private OrderTableName name;

    @Embedded
    private NumberOfGuests numberOfGuests;

    private boolean empty;

    protected OrderTable() {
    }

    public OrderTable(final OrderTableName name) {
        this.id = UUID.randomUUID();
        this.name = name;
        this.numberOfGuests = NumberOfGuests.init();
        this.empty = Boolean.TRUE;
    }

    public void sit() {
        if (!empty) {
            throw new IllegalArgumentException();
        }
        empty = Boolean.FALSE;
    }

    public NumberOfGuests getNumberOfGuests() {
        return numberOfGuests;
    }

    public boolean isEmpty() {
        return empty;
    }
}

File: src/main/java/kitchenpos/eatinorders/tobe/OrderTable.kt
package kitchenpos.eatinorders.tobe

import java.util.*

class OrderTable(
    val name: String,
) {
    init {
        require(name.isNotEmpty())
    }

    val id = UUID.randomUUID()
    var occupied: Boolean = false
    var numberOfGuests: Int = NUMBER_OF_GUEST_INITIAL_VALUE

    fun sit() {
        this.occupied = true
    }

    fun clear() {
        this.numberOfGuests = NUMBER_OF_GUEST_INITIAL_VALUE
        this.occupied = false
    }

    fun changeNumberOfGuests(numberOfGuests: Int) {
        check(numberOfGuests >= MINIMUM_NUMBER_OF_GUEST)
        this.numberOfGuests = numberOfGuests
    }

    fun canOrder() = occupied

    companion object {
        private const val MINIMUM_NUMBER_OF_GUEST = 0
        private const val NUMBER_OF_GUEST_INITIAL_VALUE = 0
    }
}
